tarok/src/game.cpp

```diff
@@ -36,24 +33,24 @@ int TarokGame::MaxChanceOutcomes() const {
 
 int TarokGame::NumPlayers() const { return num_players_; }
 
-double TarokGame::MinUtility() const {
-  // todo: implement
-  return 0.0;
-}
+double TarokGame::MinUtility() const { return -500.0; }
 
-double TarokGame::MaxUtility() const {
-  // todo: implement
-  return 0.0;
-}
+double TarokGame::MaxUtility() const { return 500.0; }
 
 std::shared_ptr<const open_spiel::Game> TarokGame::Clone() const {
-  // todo: implement
-  return nullptr;
+  return std::shared_ptr<const open_spiel::Game>(new TarokGame(*this));
 }
 
 int TarokGame::MaxGameLength() const {
-  // todo: implement
-  return 0;
+  // see tarok/python/max_game_length.py for an additional explanation about the
+  // number of actions
+  if (num_players_ == 3) {
+    // 17 actions + 16 cards each
+    return 33;
+  } else {
+    // 24 actions + 12 cards each
+    return 36;
+  }
```

Comment:
Shouldn't this be 17 + 3 * 16 = 65 for three players and 24 + 4 * 12 = 72 for four players?

Reply:
Good catch, will fix.